torch/_guards.py

```diff
@@ -72,8 +72,7 @@
 # 3. Compact: The string form is directly displayed by some tools. Special symbols are okay.
 
 
-# TODO: mark as kw_only=True once we drop support for <Python 3.10
-@dataclass(frozen=True)
+@dataclass(frozen=True, kw_only=True)
```

Comment:
Also, sicne the dataclass if frozen, why not also have it be `slots=True`?
```suggestion
@dataclass(frozen=True, kw_only=True, slots=True)